client/signup/wpcom-login-form/test/index.jsx

```diff
@@ -12,103 +14,98 @@ describe( 'WpcomLoginForm', () => {
 	};
 
 	test( 'should render default fields as expected.', () => {
-		const wrapper = shallow( <WpcomLoginForm { ...props } />, { disableLifecycleMethods: true } );
+		const { container, rerender } = render( <WpcomLoginForm { ...props } /> );
 
 		// should render root form element
-		const form = wrapper.find( 'form' );
-		expect( form ).toHaveLength( 1 );
-		expect( form.prop( 'method' ) ).toEqual( 'post' );
-		expect( form.prop( 'action' ) ).toEqual( 'https://test.wordpress.com/wp-login.php' );
+		const form = container.firstChild;
```

Comment:
Why not query the `form` element with `querySelector( 'form' )` instead of looking at `.firstChild`?

Reply:
We generally avoid using `querySelector` with `@testing-library/react`. See [this](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-container-to-query-for-elements) for context.

Reply:
I understand why `querySelector` is bad, but isn't `.firstChild` even worse? None of the two is close to how the user actually experiences the UI, and additionally with `.firstChild` it's not clear what element we are looking for, and also it breaks when the `form` element is for some reason moved to a different position.

Reply:
It depends on the point of view; from the point of view of the component developer it sometimes just makes sense to target the root element of the component, regardless of what it is. That is when we don't have a good way to use `screen` queries, of course.